ci: Add Charm CI/CD

jmuzina opened the pull request:
## Done

Part 4 (hopefully final) of #5551 

Based on #5557 , merge that first

Uses [webteam-devops](https://github.com/canonical/webteam-devops) deploy action.

## QA

- Deployment to staging must pass

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://vanilla-framework-5558.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/8160)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

jmuzina commented:
This is now [working](https://github.com/canonical/vanilla-framework/actions/runs/15890286073/job/44812462841) all the way through building and publishing both the charm and the rock to Charmhub. It fails on the Juju deployment step, because I need the credentials for our PS6 Juju models once they're created by IS.

Some things to note:
- I have added a temporary, modified copy of the webteam devops workflow that we can use until resolution of [this issue](https://github.com/canonical/webteam-devops/issues/4) that I created on their repo.
- I needed to set up a [Charmhub listing](https://charmhub.io/vanillaframework-io/) in order to publish the image and charm. Now I realize that the listing is under my name, rather than the web team's - need to find out how to create a listing with the appropriate ownership, then change the `CHARMHUB_TOKEN` secret.
